featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -737,32 +738,49 @@ def _build_transform_features(
                 current_options,
             )
 
-            # If require_direct_input, require a DirectFeature in input or as a
-            # groupby, and don't create features of inputs/groupbys which are
-            # all direct features with the same relationship path
             for matching_input in matching_inputs:
                 if not can_stack_primitive_on_inputs(groupby_prim, matching_input):
                     continue
-                if not any(
-                    True for bf in matching_input if bf.number_output_features != 1
-                ):
-                    for groupby in groupby_matches:
-                        input_features = matching_input + (groupby,)
-                        if require_direct_input and (
-                            _all_direct_and_same_path(input_features)
-                            or not any(
-                                True
-                                for feature in (input_features)
-                                if isinstance(feature, DirectFeature)
-                            )
-                        ):
-                            continue
-                        new_f = GroupByTransformFeature(
-                            list(matching_input),
-                            groupby=groupby[0],
-                            primitive=groupby_prim,
+                if require_direct_input:
```

Comment:
this block should come after the  `if any(True for bf ...)` block that is just below it